contracts/ShapeshiftDAORouter.sol

```diff
@@ -0,0 +1,435 @@
+
+        for (uint256 i = firstVaultId; i <= _lastVaultId; i++) {
+            VaultAPI vault = registry.vaults(token, i);
+            assets += vault.totalAssets();
+        }
+    }
+
+    /**
+     * @notice Called to deposit the caller's tokens into the most-current vault, crediting the minted shares to recipient.
+     * @dev The caller must approve this contract to utilize the specified ERC20 or this call will revert.
+     * @param token Address of the ERC20 token being deposited
+     * @param recipient Address to receive the issued vault tokens
+     * @param amount Amount of tokens to deposit; tokens that cannot be deposited will be refunded. If `DEPOSIT_EVERYTHING`, just deposit everything.
+     * @return Total vault shares received by recipient
+     */
+    function deposit(
+        address token,
+        address recipient,
+        uint256 amount
+    ) external returns (uint256) {
+        return
+            _deposit(
+                IERC20(token),
+                msg.sender,
+                recipient,
+                amount,
+                MAX_VAULT_ID
+            );
+    }
+
+    /**
+     * @notice Called to deposit the caller's tokens into a specific vault, crediting the minted shares to recipient.
+     * @dev The caller must approve this contract to utilize the specified ERC20 or this call will revert.
+     * @param token Address of the ERC20 token being deposited
+     * @param recipient Address to receive the issued vault tokens
+     * @param amount Amount of tokens to deposit; tokens that cannot be deposited will be refunded. If `DEPOSIT_EVERYTHING`, just deposit everything.
+     * @param vaultId Vault id to deposit into; pass `MAX_VAULT_ID` to deposit into the latest vault
+     * @return Total vault shares received by recipient
+     */
+    function deposit(
+        address token,
+        address recipient,
+        uint256 amount,
+        uint256 vaultId
+    ) external returns (uint256) {
+        return _deposit(IERC20(token), msg.sender, recipient, amount, vaultId);
+    }
+
+    /**
+     * @notice Called to deposit depositor's tokens into a specific vault, crediting the minted shares to recipient.
+     * @dev Depositor must approve this contract to utilize the specified ERC20 or this call will revert.
+     * @param token Address of the ERC20 token being deposited
+     * @param depositor Address to pull deposited funds from. SECURITY SENSITIVE.
+     * @param recipient Address to receive the issued vault tokens
+     * @param amount Amount of tokens to deposit; tokens that cannot be deposited will be refunded. If `DEPOSIT_EVERYTHING`, just deposit everything.
+     * @param vaultId Vault id to deposit into; pass `MAX_VAULT_ID` to deposit into the latest vault
+     * @return shares Total vault shares received by recipient
+     */
+    function _deposit(
+        IERC20 token,
+        address depositor,
+        address recipient,
+        uint256 amount,
+        uint256 vaultId
```

Comment:
Mostly completeness, but it's worth considering that the idea that the latest vault is always the "best" one was something I haven't seen codified outside the old BaseRouter code. I could imagine a situation where we'd want to give our users the ability to deposit into the vault with the highest APR, or the most liquid shares, or one with a larger deposit limit, or the one capable of redeeming the most shares at a time -- it just seems to me like it would be bad to introduce the assumption that latest == "best" in all dimensions at this layer.

Reply:
Hmm, okay.  I believe the vaults contain many strategies and the "best vault" is always the one that is actively maintained and most liquid.  Since we already have it, fine leaving it.